mono/metadata/assembly.c

```diff
@@ -1495,11 +1507,13 @@ load_reference_by_aname_default_asmctx (MonoAssemblyName *aname, MonoAssembly *a
 		MonoAssemblyByNameRequest req;
 		memset (&req, 0, sizeof (req));
 		req.request.asmctx = MONO_ASMCTX_DEFAULT;
```

Comment:
There seem to be a big share part of initing `MonoAssemblyByNameRequest`.
How about we extract it to a function?